src/software/ai/threaded_ai.cpp

```diff
@@ -8,7 +8,9 @@
 ThreadedAI::ThreadedAI(std::shared_ptr<const AiConfig> ai_config,
                        std::shared_ptr<const AiControlConfig> control_config,
                        std::shared_ptr<const PlayConfig> play_config)
-    : ai(ai_config, control_config, play_config), control_config(control_config)
+    : FirstInFirstOutThreadedObserver<World>(DEFAULT_BUFFER_SIZE, false),
```

Comment:
Just a minor question, what is this used for?

Reply:
I think `FirstInFirstOutThreadedObserver` used to be default constructed so it never showed up before. `FirstInFirstOutThreadedObserver<World>` is the parent class of `ThreadedAI`, i.e. `ThreadedAI` _is a_ `FirstInFirstOutThreadedObserver<World>`->it provides the AI with new World objects to process